pkg/nm/nm.go

```diff
@@ -440,3 +456,295 @@ func Devices() (map[string]json.RawMessage, error) {
 		"eth0":  eth0,
 	}, nil
 }
+//===========================VPN functions=======================
+func CheckVPNStatus()(bool, gonetworkmanager.NmVpnConnectionState,string , error){
+	nm, err := gonetworkmanager.NewNetworkManager()
+	if err!=nil {
+		return false, 0, "", fmt.Errorf("failed connecting to Network manager: %v", err)
+	}
+
+	gatewayID, err := getGatewayID()
+	if err != nil {
+		return false,0,"",fmt.Errorf("failed to get gateway ID: %v", err)
+	}
+	connected, activeConn, err := isVPNConnected(nm,gatewayID)
+	if err != nil || !connected {
+		return false,0,"", err
+	}
+
+	vpnConn, err := gonetworkmanager.NewVpnConnection(activeConn.GetPath())
+	if( err != nil || vpnConn ==nil) {
+		return true, 0, "", fmt.Errorf("failed to create VPN connection object")
+	}
+
+	state, err := vpnConn.GetPropertyVpnState()
+
+	if state == gonetworkmanager.NmVpnConnectionActivated {
+		log.Println("VPN is fully connected!")
+	}
+	if err != nil {
+		return true, 0, "", fmt.Errorf("failed to get VPN state: %v", err)
+	}
+
+	banner, err := vpnConn.GetPropertyBanner()
+	if err != nil {
+		banner = ""
+	}
+
+	return true, state, banner, nil
+}
+func EnableDisableVPN(enable bool) (error) {
+	nm, err :=gonetworkmanager.NewNetworkManager()
+	if err != nil {
+		return fmt.Errorf("failed to connect to NetworkManager: %v", err)
+	}
+	version, err := nm.GetPropertyVersion()
+	if err != nil {
+		return err
+	}
+	log.Println("[     ] Network Manager Version:", version)
+	gatewayID, err := getGatewayID()
```

Comment:
Please also pass the client ID as argument to this function. I think this file "nm.go" should NOT know anything about gateways.